lib/calendar.ts

```diff
@@ -1106,13 +1106,14 @@ abstract class HelperBase {
     return calendarDate;
   }
   addCalendar(
-    calendarDate: CalendarYMD,
+    calendarDate: CalendarYMD & { monthCode: string },
```

Comment:
Nit: is `{monthCode: string}` something that would benefit from being a named type and reused elsewhere?